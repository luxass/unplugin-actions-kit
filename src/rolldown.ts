/**
 * This entry file is for Rolldown plugin.
 *
 * @module
 */

import type unplugin from "./";
import { createRolldownPlugin } from "unplugin";
import { unpluginFactory } from "./";

/**
 * Rolldown plugin
 *
 * @example
 * ```ts
 * // rolldown.config.js
 * import actionsKit from "actions-kit/unplugin//rolldown"
 *
 * export default {
 *   plugins: [actionsKit()],
 * }
 * ```
 */
export default createRolldownPlugin(unpluginFactory) as typeof unplugin.rolldown;
