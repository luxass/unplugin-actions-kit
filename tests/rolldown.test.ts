import { readdir, readFile } from "node:fs/promises";
import { join } from "node:path";
import { rolldown } from "rolldown";
import { describe, expect, it } from "vitest";
import { fromFileSystem, testdir } from "vitest-testdirs";
import ActionKitPlugin from "../src/rolldown";

it("expect no `actions-kit.d.ts` file generated if plugin not in use", async () => {
  const directoryJson = await fromFileSystem("./tests/fixtures/basic");
  const testdirPath = await testdir(directoryJson);

  expect(testdirPath).toBeDefined();

  const bundle = await rolldown({
    input: join(testdirPath, "index.ts"),
    platform: "node",
    external: ["@actions/core"],
  });

  const { output } = await bundle.generate({
    format: "cjs",
    sourcemap: false,
  });

  expect(output[0]).toBeDefined();
  expect(output[0].code).toBeDefined();
  expect(output[0].code).toMatchSnapshot();

  const files = await readdir(testdirPath);
  expect(files).not.toContain("actions-kit.d.ts");
});

it("expect `actions-kit.d.ts` to be generated", async () => {
  const directoryJson = await fromFileSystem("./tests/fixtures/basic");
  const testdirPath = await testdir(directoryJson);

  expect(testdirPath).toBeDefined();

  const bundle = await rolldown({
    input: join(testdirPath, "index.ts"),
    platform: "node",
    external: ["@actions/core"],
    plugins: [
      ActionKitPlugin({
        actionPath: join(testdirPath, "action.yaml"),
      }),
    ],
  });

  const { output } = await bundle.generate({
    format: "cjs",
    sourcemap: false,
  });

  const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

  expect(output[0]).toBeDefined();
  expect(output[0].code).toBeDefined();
  expect(output[0].code).toMatchSnapshot();
  expect(dtsOutput).toMatchSnapshot();
});

describe("augmentations", () => {
  it("expect only inputs to be augmented", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/only-inputs");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).not.toContain("ActionOutputName");
    expect(dtsOutput).toContain("ActionInputName");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });

  it("expect only outputs to be augmented", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/only-outputs");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");

    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).toContain("ActionOutputName");
    expect(dtsOutput).not.toContain("ActionInputName");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });

  it("expect no augmentations", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/empty");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).not.toContain("ActionOutputName");
    expect(dtsOutput).not.toContain("ActionInputName");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });
});

describe("inject", () => {
  it("expect `ACTION_INPUTS` in global scope", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/only-inputs");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
          inject: "inputs",
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).not.toContain("ActionOutputName");
    expect(dtsOutput).not.toContain("ACTION_OUTPUTS");
    expect(dtsOutput).toContain("ActionInputName");
    expect(dtsOutput).toContain("ACTION_INPUTS");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });

  it("throw if missing `inputs`", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/only-outputs");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
          inject: "inputs",
        }),
      ],
    });

    expect(async () => await bundle.generate({
      format: "cjs",
      sourcemap: false,
    })).rejects.toThrow("inputs is not defined in action.yml");
  });

  it("expect `ACTION_OUTPUTS` in global scope", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/only-outputs");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
          inject: "outputs",
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).toContain("ActionOutputName");
    expect(dtsOutput).toContain("ACTION_OUTPUTS");
    expect(dtsOutput).not.toContain("ActionInputName");
    expect(dtsOutput).not.toContain("ACTION_INPUTS");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });

  it("throw if missing `outputs`", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/only-inputs");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
          inject: "outputs",
        }),
      ],
    });

    expect(async () => await bundle.generate({
      format: "cjs",
      sourcemap: false,
    })).rejects.toThrow("outputs is not defined in action.yml");
  });

  it("expect `ACTION_INPUTS` & `ACTION_OUTPUTS` in global scope", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/basic");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
          inject: true,
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).toContain("ActionOutputName");
    expect(dtsOutput).toContain("ACTION_OUTPUTS");
    expect(dtsOutput).toContain("ActionInputName");
    expect(dtsOutput).toContain("ACTION_INPUTS");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });

  it("expect no change in global scope", async () => {
    const directoryJson = await fromFileSystem("./tests/fixtures/basic");
    const testdirPath = await testdir(directoryJson);

    expect(testdirPath).toBeDefined();

    const bundle = await rolldown({
      input: join(testdirPath, "index.ts"),
      platform: "node",
      external: ["@actions/core"],
      plugins: [
        ActionKitPlugin({
          actionPath: join(testdirPath, "action.yaml"),
          inject: false,
        }),
      ],
    });

    const { output } = await bundle.generate({
      format: "cjs",
      sourcemap: false,
    });

    const dtsOutput = await readFile(join(testdirPath, "actions-kit.d.ts"), "utf-8");
    expect(output[0]).toBeDefined();
    expect(output[0].code).toBeDefined();

    expect(dtsOutput).toContain("ActionOutputName");
    expect(dtsOutput).not.toContain("ACTION_OUTPUTS");
    expect(dtsOutput).toContain("ActionInputName");
    expect(dtsOutput).not.toContain("ACTION_INPUTS");

    expect(dtsOutput).toMatchSnapshot();
    expect(output[0].code).toMatchSnapshot();
  });
});

it("custom output path", async () => {
  const directoryJson = await fromFileSystem("./tests/fixtures/basic");
  const testdirPath = await testdir(directoryJson);

  expect(testdirPath).toBeDefined();

  const bundle = await rolldown({
    input: join(testdirPath, "index.ts"),
    platform: "node",
    external: ["@actions/core"],
    plugins: [
      ActionKitPlugin({
        actionPath: join(testdirPath, "action.yaml"),
        outputPath: join(testdirPath, "custom"),
      }),
    ],
  });

  const { output } = await bundle.generate({
    format: "cjs",
    sourcemap: false,
  });

  const dtsOutput = await readFile(join(testdirPath, "custom", "actions-kit.d.ts"), "utf-8");

  expect(output[0]).toBeDefined();
  expect(output[0].code).toBeDefined();
  expect(output[0].code).toMatchSnapshot();
  expect(dtsOutput).toMatchSnapshot();
});
