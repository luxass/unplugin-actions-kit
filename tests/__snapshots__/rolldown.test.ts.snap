// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`augmentations > expect no augmentations 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {


}
"
`;

exports[`augmentations > expect no augmentations 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-augmentations-expect-no-augmentations/index.ts
async function run() {
	core.info("hello world!");
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`augmentations > expect only inputs to be augmented 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;


}
"
`;

exports[`augmentations > expect only inputs to be augmented 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-augmentations-expect-only-inputs-to-be-augmented/index.ts
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`augmentations > expect only outputs to be augmented 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`augmentations > expect only outputs to be augmented 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-augmentations-expect-only-outputs-to-be-augmented/index.ts
async function run() {
	core.setOutput("message", "Hello, World!");
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`custom output path 1`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-custom-output-path/index.ts
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
	core.setOutput("message", message);
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`custom output path 2`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`expect \`actions-kit.d.ts\` to be generated 1`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-expect-actions-kit-d-ts-to-be-generated/index.ts
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
	core.setOutput("message", message);
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`expect \`actions-kit.d.ts\` to be generated 2`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`expect no \`actions-kit.d.ts\` file generated if plugin not in use 1`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-expect-no-actions-kit-d-ts-file-generated-if-plugin-not-in-use/index.ts
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
	core.setOutput("message", message);
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`inject > expect \`ACTION_INPUTS\` & \`ACTION_OUTPUTS\` in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";

declare global {
  export const ACTION_INPUTS = {
    "type": "type",
    "who-to-greet": "who-to-greet",
  };

  export const ACTION_OUTPUTS = {
    "message": "message",
  };

}

declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`inject > expect \`ACTION_INPUTS\` & \`ACTION_OUTPUTS\` in global scope 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-inject-expect-ACTION_INPUTS-ACTION_OUTPUTS-in-global-scope/index.ts
globalThis.ACTION_INPUTS = {
	"type": "type",
	"who-to-greet": "who-to-greet"
};
globalThis.ACTION_OUTPUTS = { "message": "message" };
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
	core.setOutput("message", message);
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`inject > expect \`ACTION_INPUTS\` in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";

declare global {
  export const ACTION_INPUTS = {
    "type": "type",
    "who-to-greet": "who-to-greet",
  };

}

declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;


}
"
`;

exports[`inject > expect \`ACTION_INPUTS\` in global scope 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-inject-expect-ACTION_INPUTS-in-global-scope/index.ts
globalThis.ACTION_INPUTS = {
	"type": "type",
	"who-to-greet": "who-to-greet"
};
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`inject > expect \`ACTION_OUTPUTS\` in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";

declare global {
  export const ACTION_OUTPUTS = {
    "message": "message",
  };

}

declare module "@actions/core" {

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`inject > expect \`ACTION_OUTPUTS\` in global scope 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-inject-expect-ACTION_OUTPUTS-in-global-scope/index.ts
globalThis.ACTION_OUTPUTS = { "message": "message" };
async function run() {
	core.setOutput("message", "Hello, World!");
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;

exports[`inject > expect no change in global scope 1`] = `
"/* eslint-disable */
// @ts-nocheck
// generated by 'actions-kit'

import type * as core from "@actions/core";



declare module "@actions/core" {

  type ActionInputName = "type" | "who-to-greet";

  export function getInput(name: ActionInputName, options?: core.InputOptions): string;

  type ActionOutputName = "message";

  export function setOutput(name: ActionOutputName, value: any): void;


}
"
`;

exports[`inject > expect no change in global scope 2`] = `
""use strict";
//#region rolldown:runtime
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __copyProps = (to, from, except, desc) => {
	if (from && typeof from === "object" || typeof from === "function") for (var keys = __getOwnPropNames(from), i = 0, n = keys.length, key; i < n; i++) {
		key = keys[i];
		if (!__hasOwnProp.call(to, key) && key !== except) __defProp(to, key, {
			get: ((k) => from[k]).bind(null, key),
			enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable
		});
	}
	return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", {
	value: mod,
	enumerable: true
}) : target, mod));

//#endregion
const core = __toESM(require("@actions/core"));

//#region .vitest-testdirs/vitest-rolldown-inject-expect-no-change-in-global-scope/index.ts
async function run() {
	const type = core.getInput("type");
	const whoToGreet = core.getInput("who-to-greet");
	const message = \`Hello \${whoToGreet}!\`;
	core.info(message);
	if (type === "error") {
		core.setFailed("This is an error!");
	}
	core.setOutput("message", message);
}
run().catch((err) => {
	console.error(err);
	core.setFailed(err.message);
});

//#endregion"
`;
